tower-posix/src/Ivory/OS/Posix/Tower/Signal.hs

```diff
@@ -18,7 +18,7 @@ instance Signalable Watcher where
   signalName = watcherName
   signalHandler = watcherDefs
   signalInit = watcherInit
-  signalNumber = error "no signal numbers defined for Watcher"
+  signalNumber _ = 42424242
```

Comment:
Long story ... 
This code was always buggy, but it was never executed because we dropped the AST. Now it is not the case anymore, hence the error is evaluated even if we never use it anywhere, which results in a systematic crash. I changed it to 42424242 to check if this value appears anywhere is the C source code, which is not the case.

Ideally this should be changed to a Maybe Int (and here into Nothing), but there are lot of modifications to do. I'll probably do it soon before merging the pull request.